### DISCUSSION TITLE
Optional Select TValue

### DISCUSSION THREAD

**Discussion by Zumpel96** [Megabit/Blazorise#3833] [Q&A]
Hello. I have a Select, which contains a List of Companies. Anyway, From the EF, I therefore have a nullable Property `int? CompanyId`, because I want that value to be optional. I need this to be `null` and not `0`, because otherwise EF will look for a reference of the ID `0`, which is not contained within my database and throw an exception. How can I depict this properly in Blazorise? If I set the `TValue` of the Select to `int?` everything works, but I need an element in the List, with the value `null`. Therefore, the following code does not work:

```
<Select TValue="int?" @bind-SelectedValue="@Data.CompanyId">
    <SelectItem Value="@null">@L["NoSelectedCompany"]</SelectItem>
    @foreach (var company in companies)
    {
        <SelectItem Value="@company.Id">@company.Name</SelectItem>
    }
</Select>
```

**Comment by stsrki** [comment]
You need to define `TValue` so that it matches with `TValue` from the parent Select. This might also work `<SelectItem Value="@((int?)null)">@L["NoSelectedCompany"]</SelectItem>`.

**Comment by Zumpel96** [comment]
Thanks, with the cast it works. Anyways, it seems, that the `SelectedValueChanged` is not triggered. To test it, I changed my code to:

```
<Select TValue="int?" SelectedValue="@Data.CompanyId" SelectedValueChanged="TestValueChanged">
    <SelectItem TValue="int?" Value="null">@L["NoSelectedCompany"]</SelectItem>
    @foreach (var company in companies)
    {
        <SelectItem Value="@company.Id">@company.Name</SelectItem>
    }
</Select>

public void TestValueChanged(int? i)
{
    this.Data.CompanyId = i;
}
```

If I set a breakpoint into the line `this.Data.CompanyId = i` it gets triggered when a non-null value is set. When I select the null-value, the breakpoint does not trigger. I also don't get any additional warnings or errors.

**Reply by stsrki** [reply]
The null or empty `option` element has a strange behavior. It is a known issue, and it is how it works on browsers. You can try searching on our issues list for more info, eg. https://github.com/Megabit/Blazorise/issues/185

In the meantime try with `<SelectItem TValue="int?" Value="null"></SelectItem>`

**Reply by Zumpel96** [reply]
Oh I see. Thanks for the issue list. Sadly `<SelectItem TValue="int?" Value="null">@L["NoSelectedCompany"]</SelectItem>` has the same behavior. I'll try to find a workaround until then.

**Reply by stsrki** [reply]
Did you remove `@L["NoSelectedCompany"]` from the SelectItem content?
